### PR TITLE
Correctly unprotect worksheet without password

### PR DESCRIPTION
**Pull request by igitur**
This PR partially addresses #866 . It fixes a bug with unprotected worksheets that are protected without a password.

Worksheets protected with the SHA512 algorithm are not supported yet.

### CODE REVIEW

File: ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        [Test]
+        public void TestUnprotectWorksheetWithNoPassword()
+        {
+            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Misc\SHA512PasswordProtection.xlsx")))
+            using (var wb = new XLWorkbook(stream))
+            {
+                wb.Worksheet("Sheet1").Unprotect();
+
+                // Protected with SHA-512 password - not yet supported.
```
Comment (Pankraty) [inline ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs]: I think this test should be split in two: the first unprotecting `Sheet1` and the second unprotecting `Sheet2`. Until SHA-512 is implemented the second one must be muted, but both must have correct assertions from the beginning (e.g. `Assert.IsFalse(ws.Protection.Protected);`).

File: ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                Assert.Throws<ArgumentException>(() => wb.Worksheet("Sheet2").Unprotect());
```
Comment (Pankraty) [inline ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs]: `NotSupportedException` or `NotImplementedException` would be more informative.

Comment (igitur) [reply ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs]: That would be a breaking change. I'd rather spend any effort on actually implementing this algorithm.

File: ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs (hunk 1 above)
Comment (igitur) [reply ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs]: Ok, the split them in to, but note that the SHA-512 tests is just a temporary measure until we actually implement the algorithm.

**Pull request merged**